### PR TITLE
Allow passthrough of variable Negations if not present

### PR DESCRIPTION
**Pull request by piyushroshan**
> Thank you for contributing to Coraza WAF, your effort is greatly appreciated
> Before submitting check if what you want to add to `coraza` list meets [quality standards](https://github.com/corazawaf/coraza/blob/master/CONTRIBUTING.md#quality-standards) before sending pull request. Thanks!

Allow passthrough of variable Negations if not present #264 

**Note**: _that go.mod and go.sum can only be modified for tested dependency updates or justified new features._

**Make sure that you've checked the boxes below before you submit PR:**

- [x] My code includes positive and negative tests.
- [x] I have an appropriate description with correct grammar.
- [x] I have read [Contribution guidelines](https://github.com/corazawaf/coraza/blob/master/CONTRIBUTING.md#contribution-guidelines), [maintainers note](https://github.com/corazawaf/coraza/blob/master/CONTRIBUTING.md#maintainers) and [Quality standard](https://github.com/github.com/corazawaf/coraza/v2sso/coraza-waf/blob/master/CONTRIBUTING.md#quality-standards).

Thanks for your PR :heart:

### CODE REVIEW

**Review by jptosso — DISMISSED** [review]
counter is not being used

**Review by bxlxx — COMMENTED** [review]
Everything is fine but I have some observations that I would like to discuss.

File: rule.go [hunk 1]
```diff
@@ -441,29 +441,29 @@ func (r *Rule) AddVariable(v variables.RuleVariable, key string, iscount bool) e
...
 func (r *Rule) AddVariableNegation(v variables.RuleVariable, key string) error {
-	counter := 0
 	var re *regexp.Regexp
 	if len(key) > 2 && key[0] == '/' && key[len(key)-1] == '/' {
 		key = key[1 : len(key)-1]
 		re = regexp.MustCompile(key)
 	}
-
+	// Prevent sigsev
+	if r == nil {
```
[the remark this answers is not in the document]
Comment (piyushroshan) [reply rule.go:458]: In case of SecRuleUpdateTargetByID
```

**Review by bxlxx — APPROVED** [review]
LGTM

**Pull request merged**